crates/aptos-rosetta/src/main.rs

```diff
@@ -48,24 +51,25 @@ async fn main() {
 
         // Wait and ensure the node is running on the URL
         let client = aptos_rest_client::Client::new(online_args.rest_api_url.clone());
-        let mut successful = false;
-        let total_wait_duration = Duration::from_secs(TOTAL_REST_API_WAIT_DURATION_S);
         let start = Instant::now();
-        while start.elapsed() < total_wait_duration {
-            if client.get_index_bcs().await.is_ok() {
-                successful = true;
-                break;
+        loop {
+            match client.get_index_bcs().await {
+                Ok(_) => {
+                    break;
+                }
+                Err(err) => {
+                    sample!(
+                        SampleRate::Duration(Duration::from_millis(LOG_INTERVAL_MS)),
+                        println!(
+                            "aptos-rosetta: Full node REST API isn't responding yet.  You should check the node logs.  It's been waiting {} seconds.  Error: {:?}",
+                            start.elapsed().as_secs(),
+                            err
+                        )
+                    );
+                    tokio::time::sleep(Duration::from_millis(DEFAULT_REST_API_WAIT_INTERVAL_MS))
+                        .await;
+                }
             }
-
-            tokio::time::sleep(Duration::from_millis(REST_API_WAIT_DURATION_MS)).await;
-        }
-
-        // If it didn't start up, we need to crash
-        if !successful {
-            panic!(
-                "aptos-rosetta: Local full node didn't start up on time after {} seconds at {}",
-                TOTAL_REST_API_WAIT_DURATION_S, online_args.rest_api_url
-            )
```

Comment:
why does the logger come into place so late? why use println instead of eprintln?